packages/editor/src/utils/media-upload/test/media-upload.js

```diff
@@ -29,19 +31,72 @@ describe( 'mediaUpload', () => {
 		mediaUpload( {
 			filesList: [ invalidMediaObj ],
 			onFileChange: onFileChangeSpy,
-			allowedType: 'image',
+			allowedTypes: [ 'image' ],
 			onError,
 		} );
 		expect( onFileChangeSpy ).not.toHaveBeenCalled();
 		expect( onError ).toHaveBeenCalled();
 		expect( onError.mock.calls[ 0 ][ 0 ].code ).toBe( 'MIME_TYPE_NOT_SUPPORTED' );
 	} );
 
+	it( 'should correctly validate an allowedType containing a multiple type string', () => {
+		// When the upload has success we get errors because we don't have available on the test environment
+		// all the functions required by mediaUpload.
+		// We know that when a validation success the function tries to creat a temporary url for the file
+		// so we use that function to test the validation success.
+
+		const onError = jest.fn();
+		const testFile = {
+			url: 'https://cldup.com/uuUqE_dXzy.mp3',
+			type: 'audio/mp3',
+			name: 'mymusic.mp3',
+		};
+		expect( () => {
+			mediaUpload( {
+				filesList: [ testFile ],
+				onFileChange: onFileChangeSpy,
+				allowedTypes: [ 'video', 'image' ],
+				onError,
+			} );
+		} ).not.toThrow();
+		expect( createObjectURL ).not.toHaveBeenCalled();
+		expect( onError ).toHaveBeenCalled();
+		expect( onError.mock.calls[ 0 ][ 0 ].code ).toBe( 'MIME_TYPE_NOT_SUPPORTED' );
+
+		createObjectURL.mockReset();
+		onError.mockReset();
+
+		expect( () => {
```

Comment:
I'd usually expect these to be two separate unit tests with the expected outcome described in the test case description. That would make it easier to pin down the problem should one of them fail.